auction/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::result;
+use sr_primitives::traits::{MaybeSerializeDeserialize, Member, SimpleArithmetic};
+use srml_support::{
+	decl_error, decl_event, decl_module, decl_storage, dispatch::Result, ensure, Parameter, StorageMap, StorageValue,
+};
+use srml_system::{self as system, ensure_signed};
+use utilities::{LinkedItem, LinkedList};
+
+use traits::{Auction, AuctionHandler, AuctionInfo, OnNewBidResult};
+
+mod mock;
+mod tests;
+
+pub trait Trait: srml_system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as srml_system::Trait>::Event>;
+	type Balance: Parameter + Member + SimpleArithmetic + Default + Copy + MaybeSerializeDeserialize;
+	type AuctionId: Parameter + Member + SimpleArithmetic + Default + Copy + MaybeSerializeDeserialize;
+	type Handler: AuctionHandler<Self::AccountId, Self::Balance, Self::BlockNumber, Self::AuctionId>;
+}
+
+type AuctionEndTimeList<T> =
+	LinkedList<AuctionEndTime<T>, <T as srml_system::Trait>::BlockNumber, <T as Trait>::AuctionId>;
+
+decl_event!(
+	pub enum Event<T> where
+		<T as srml_system::Trait>::AccountId,
+		<T as Trait>::Balance,
+		<T as Trait>::AuctionId,
+	{
+		Bid(AuctionId, AccountId, Balance),
+	}
+);
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Auction {
+		pub Auctions get(fn auctions): map T::AuctionId => Option<AuctionInfo<T::AccountId, T::Balance, T::BlockNumber>>;
+		pub AuctionsCount get(fn auctions_count) build(|_| 0.into()): T::AuctionId;
```

Comment:
```suggestion
		pub AuctionsCount get(fn auctions_count): T::AuctionId;
```
It will use `AuctionId::default()` which will give 0